src/python-common/ceph/deployment/drive_group.py

```diff
@@ -123,29 +190,70 @@ def __init__(self,
         self.osd_id_claims = osd_id_claims
 
     @classmethod
-    def from_json(self, json_drive_group):
+    def from_json(cls, json_drive_group):
+        # type: (dict) -> DriveGroupSpec
         """
         Initialize 'Drive group' structure
 
         :param json_drive_group: A valid json string with a Drive Group
                specification
         """
-        args = {k: (DeviceSelection.from_json(v) if k.endswith('_devices') else v) for k, v in
-                json_drive_group.items()}
-        return DriveGroupSpec(**args)
+        for applied_filter in list(json_drive_group.keys()):
+            if applied_filter not in cls._supported_features:
+                raise DriveGroupValidationError(
+                    "Feature <{}> is not supported".format(applied_filter))
+
+        for key in ('block_wal_size', 'block_db_size', 'journal_size'):
+            if key in json_drive_group:
+                if isinstance(json_drive_group[key], six.string_types):
+                    from ceph.deployment.drive_selection import SizeMatcher
+                    json_drive_group[key] = SizeMatcher.str_to_byte(json_drive_group[key])
+
+        try:
+            args = {k: (DeviceSelection.from_json(v) if k.endswith('_devices') else v) for k, v in
+                    json_drive_group.items()}
+            return DriveGroupSpec(**args)
+        except (KeyError, TypeError) as e:
+            raise DriveGroupValidationError(str(e))
 
     def hosts(self, all_hosts):
+        # type: (List[str]) -> List[str]
         return fnmatch.filter(all_hosts, self.host_pattern)
 
     def validate(self, all_hosts):
```

Comment:
I would call this method when a new DriveGroupSpec object is created (via init or from_json) .

Reply:
agree